Bottom: 8b7c068f5e3f4959180c457cc431944c0d369b03
Top:    54167cb5ff59fcb0ac40b79699b2bdf50249a074
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 22:58:50 +0100

Refresh of set-c-convert-some-explicit-ma

---

diff --git a/src/action.c b/src/action.c
index eddbbbd..7744cd1 100644
--- a/src/action.c
+++ b/src/action.c
@@ -4027,24 +4027,27 @@ ActionMarkCrosshair (int argc, char **argv, int x, int y)
     {
       if (Marked.status)
 	{
-	  DrawMark ();
+	  notify_mark_change (false);
 	  Marked.status = false;
+	  notify_mark_change (true);
 	}
       else
 	{
+	  notify_mark_change (false);
+	  Marked.status = false;
 	  Marked.status = true;
 	  Marked.X = Crosshair.X;
 	  Marked.Y = Crosshair.Y;
-	  DrawMark ();
+	  notify_mark_change (true);
 	}
     }
   else if (GetFunctionID (function) == F_Center)
     {
-      DrawMark ();
+      notify_mark_change (false);
       Marked.status = true;
       Marked.X = Crosshair.X;
       Marked.Y = Crosshair.Y;
-      DrawMark ();
+      notify_mark_change (true);
     }
   return 0;
 }
